### PR TITLE
Classes associations

### PR DESCRIPTION
**Pull request by sebalp100**
# Classes associations

For this milestone I added the following:

- New Classroom, Rental, and Book classes
- Has-many/belongs-to relationship between Classroom and Student
- Many-to-many relationship between Person and Book using the intermediate class Rental

### CODE REVIEW

File: classroom.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  attr_reader :students
+
+  def initialize(label)
+    @label = label
+    @students = []
+  end
+
+  def add_student(student)
+    @students.push(student)
+    student.label = self
```
Comment (jerryowusu) [inline classroom.rb]: This is suppose to create associations between students and classrooms, so it should be `student.classroom = self` instead. :wink:

File: student.rb [hunk 2]
```diff
@@ -1,13 +1,20 @@
...
 
   def initialize(name, age, classroom, parent_permission: true)
     super(name, age, parent_permission: parent_permission)
     @classroom = classroom
   end
 
+  def label=(label)
+    @label = label
+    label.students.push(self) unless label.students.include?(self)
+  end
```
Comment (jerryowusu) [inline student.rb]: Here, remember that the relationship should be between students and classrooms and not labels. So kindly update it accordingly. :+1:

File: book.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rental(rental)
+    @rentals.push(rental)
+  end
```
Comment (jerryowusu) [inline book.rb]: As part of the project requirements, kindly make sure the add_rental method creates a new rental (like Rental.new) with the person and date as parameters. The book itself should also be sent as a parameter to create the new rental to help achieve the 'has-many' association.
Kindly check [this link](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship) for more information on that.

File: person.rb [hunk 4]
```diff
@@ -10,6 +10,11 @@ def initialize(age, name = 'Unknown', parent_permission: true)
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rentals = []
+  end
+
+  def add_rental(rental)
+    @rentals.push(rental)
   end
```
Comment (jerryowusu) [inline person.rb]: As part of the project requirements, kindly make sure the add_rental method creates a new rental (like Rental.new) with the book and date as parameters. The person itself should also be sent as a parameter to create the new rental to help achieve the 'has-many' association.
Kindly check [this link](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship) for more information on that. :pray:

**Review by jerryowusu — CHANGES_REQUESTED** [review]
Hi @sebalp100  

Good job so far!
Your project is looking fantastic so far :star_struck: 

However, there are few changes you need to make in order to move on to the next level :sunglasses:

almost there


## Highlights
Great effort so far :+1: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by sebalp100** [comment]
Changes required implemented

**Review by Kossi-stack — APPROVED** [review]
## Status: Approved ✔️

**Hi @sebalp100,**

You did a really great job! 💯 💯💯💯
Congratulations! 🎉 🎉 🎉 🎉 

128382030-38e7de91-f6f2-45cf-b54e-8068e0446e4c

You may close this Pull Request and merge the changes
Keep being awesome dev's and good luck with your next milestone!

## To Highlight

✅ Descriptive pull request message
✅ linter error checks validated
✅ Documented your work professionally
✅ README file is descriptive.
✅ Commit messages are meaningful.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding! 👏 👏 👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

**Pull request merged**